pkg/modulewriter/tfwriter.go

```diff
@@ -186,46 +186,23 @@ func writeMain(
 	return writeHclFile(filepath.Join(dst, "main.tf"), hclFile)
 }
 
-type provider struct {
-	alias   string
-	source  string
-	version string
-	config  config.Dict
-}
-
-func getProviders(bp config.Blueprint) []provider {
-	gglConf := config.Dict{}
-	for s, v := range map[string]string{
-		"project": "project_id",
-		"region":  "region",
-		"zone":    "zone"} {
-		if bp.Vars.Has(v) {
-			gglConf = gglConf.With(s, config.GlobalRef(v).AsValue())
-		}
-	}
-
-	return []provider{
-		{"google", "hashicorp/google", ">= 4.84.0, < 5.32.0", gglConf},
-		{"google-beta", "hashicorp/google-beta", ">= 4.84.0, < 5.32.0", gglConf},
-	}
-}
-
-func writeProviders(providers []provider, dst string) error {
+func writeProviders(providers map[string]config.TerraformProvider, dst string) error {
 	hclFile := hclwrite.NewEmptyFile()
 	hclBody := hclFile.Body()
 
-	for _, prov := range providers {
+	for _, k := range orderKeys[config.TerraformProvider](providers) {
```

Comment:
AFAIK `[config.TerraformProvider]` can be removed.

Reply:
I was seeing syntax errors when I did that.